charts/velero/templates/deployment.yaml

```diff
@@ -218,7 +218,7 @@ spec:
               valueFrom:
                 secretKeyRef:
                   name: {{ include "velero.secretName" $ }}
-                  key: {{ default "none" $value }}
+                  key: {{ default "none" $key }}
```

Comment:
@k4mrul Can you help check this? Because it's the revert of your change #425

Reply:
`a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name',  or 'KEY_NAME',  or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+')`

Possibly, the key name is wrong. Here is the example of usage

`secret.yaml`
```
apiVersion: v1
kind: Secret
metadata:
    name: velero-dospace-secret
    namespace: velero
stringData:
    dovalue: <do-token>
    cloud: |-
        [default]
        aws_access_key_id=<key>
        aws_secret_access_key=<secret>
```

`values.yaml`
```
credentials:
  existingSecret: velero-dospace-secret
  secretContents: {}

  extraEnvVars: 
    DIGITALOCEAN_TOKEN: dovalue
```

Now if we apply secrets.yaml and use the above values.yaml in velero, the pod will mount key and value from secret.

```
    - name: DIGITALOCEAN_TOKEN
      valueFrom:
        secretKeyRef:
          key: dovalue
          name: velero-dospace-secret
```

I just tested this config with chart verion 3.0.0. Working fine.

Reply:
There should be a note that the secret key has to be named `cloud`?

Reply:
Thanks @k4mrul for the clarification. should have read the updated values.yaml more carefully as I was expecting `credentials.extraEnvVars` behaves the same as e.g. `nodeAgent.extraEnvVars`.